src/wordvec.c

```diff
@@ -0,0 +1,50 @@
+#define _GNU_SOURCE
+#include "wordvec.h"
+
+struct wordvec *wordvec_new(void)
+{
+	struct wordvec *wv = malloc(sizeof(*wv));
+	if (!wv)
+		return NULL;
+
+	wv->len = 0;
+	wv->cap = WORDVEC_DEFAULT_CAP;
+
+	wv->vec = calloc(wv->cap, sizeof(*wv->vec));
+	if (!wv->vec) {
+		free(wv);
+		return NULL;
+	}
+
+	return wv;
+}
+
+void wordvec_del(struct wordvec *wv)
+{
+	free(wv->vec);
+	free(wv);
+}
+
+
+static struct wordvec *double_capacity(struct wordvec *wv)
+{
+	wv->cap *= 2;
+
+	wv->vec = reallocarray(wv->vec, wv->cap, sizeof(*wv->vec));
+	if (!wv->vec)
+		return NULL;
```

Comment:
If ``reallocarray`` fails, ``wv->vec`` still need to be ``free`` (``man 3 reallocarray``):
> If such an overflow occurs,  reallocarray()  returns  NULL,  sets  errno  to ENOMEM, and leaves the original block of memory unchanged.

Reply:
3318f1d fixes not losing the pointer to the original allocation.
A function catches the return value can then choose itself whether to free the wordvec or not.